### ISSUE TITLE
RLS blocking upgrades, but RLS not installed?

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rustup#1626]
Not sure how to proceed here:

```
> rustup update nightly
info: syncing channel updates for 'nightly-x86_64-apple-darwin'
info: latest update on 2019-01-28, rust version 1.34.0-nightly (da6ab956e 2019-01-27)
error: component 'rls' for target 'x86_64-apple-darwin' is unavailable for download
> rustup component remove rls
error: toolchain 'nightly-x86_64-apple-darwin' does not contain component 'rls' for target 'x86_64-apple-darwin'
```

Let me know what information could be useful!

**Comment by h-michael** [comment]
Could you try removing nigtghly at once and retry install nightly?

**Comment by nikomatsakis** [comment]
```
> rustup uninstall nightly
info: uninstalling toolchain 'nightly-x86_64-apple-darwin'
rustup updainfo: toolchain 'nightly-x86_64-apple-darwin' uninstalled
> rustup update nightly
info: syncing channel updates for 'nightly-x86_64-apple-darwin'
info: latest update on 2019-01-29, rust version 1.34.0-nightly (d8a0dd7ae 2019-01-28)
info: downloading component 'rustc'
 79.2 MiB /  79.2 MiB (100 %)  10.7 MiB/s ETA:   0 s
info: downloading component 'rust-std'
 50.9 MiB /  50.9 MiB (100 %)  10.9 MiB/s ETA:   0 s
info: downloading component 'cargo'
info: downloading component 'rust-docs'
info: installing component 'rustc'
info: installing component 'rust-std'
info: installing component 'cargo'
info: installing component 'rust-docs'

  nightly-x86_64-apple-darwin installed - rustc 1.34.0-nightly (d8a0dd7ae 2019-01-28)
```

**Comment by kinnison** [comment]
I'd like, if I may, to summarise what I *think* happened.  If it matches reality then I'll begin to chase down a root cause:

@nikomatsakis If you could have a quick look at this sequence and either correct it or let me know it's right, and also answer the bits below, that'd be awesome.

1. You had a nightly-x86_64-apple-darwin installed (presumably on your mac given it appears to be your default host)
2. You had the `rls` component installed in that toolchain
3. You ran `rustup update nightly`
4. Rustup proceeded to sync the channel, but then stopped because the new nightly lacked `rls`
5. You decided to remove `rls` so you could upgrade
6. You ran `rustup component remove rls`
7. Rustup decided to tell you that you didn't have `rls` installed

Assuming you stopped exactly there, and that before you tried the remove/reinstall suggested above you took a backup of that nightly directory, could you also confirm what happens if you run `rustup update nightly` *again*

If I'm super-lucky, you happen to know the exact nightly date you had installed, and also the nightly you tried to fetch, so we can examine the manifests too.

Sorry for the big questions, but sadly my initial hunch didn't bear fruit so I need to think harder, and evidence/data always helps with that :-D

**Comment by nikomatsakis** [comment]
Sadly, I did not make any backups. That series of steps sounds correct. One note: I think the rls component was installed by the RLS plugin.

**Comment by h-michael** [comment]
FYI: @kinnison  
I found similar issue on rls issue board. https://github.com/rust-lang/rls/issues/1273
I had poster to leave some information.
I hope this will help.

**Comment by kupiakos** [comment]
I can also confirm I tried to install it with the RLS VSCode plugin and it did not work.

**Comment by h-michael** [comment]
@kupiakos 
Thanks for your additional information.
I'll also see vscode-plugin later :)

**Comment by norcalli** [comment]
I experienced the same problems and uninstalling and reinstalling the nightly chain worked for me.

**Comment by kinnison** [comment]
@norcalli I don't suppose you kept copies of your nightly directory before/after the upgrade attempt and then after the reinstall so that I can look for differences?

**Comment by norcalli** [comment]
@kinnison Nope, sorry, but I would recommend typing up detailed instructions for the next person to stumble upon this issue through search since I'm not exactly sure what you want right now. Step by step would be good.

**Comment by kinnison** [comment]
Unfortunately @norcalli if someone didn't think to take a backup before doing the update, the initial state would be missing from any investigation, making it much harder to determine why the update failed.  I am slowly working my way through the code and if I can work out how this might occur, I may have hopes of being able to fix it :D
